trunk/modules/python/dataset/ingest/labeled.py

```diff
@@ -20,6 +27,11 @@ def readAndDivideData(path, holdoutPercentage, minTest=5, log=None) :
                 log.debug('No files found in [' + root + ']')
             continue
 
+        # find the common file type for the directory
+        # and exclude any files not matching it
+        suffix = mostCommon(files, lambda f: os.path.splitext(f)[1])
+        files = [f for f in files if f.endswith(suffix)]

```

Comment:
Okay, after speaking, maybe do a quick speed test against this and filter(). If this is faster, than use it. I mostly want faster execution to readability. I still find this very legible.